packages/server/src/mapping/tunnels/service.ts

```diff
@@ -90,6 +117,23 @@ export class TunnelService extends Service {
     }
   }
 
+  private async getByClientAndCustomChannel(clientId: uuid, customChannelName: string): Promise<Tunnel | undefined> {
+    const cached = this.cacheByClientAndCustomChannel.get(clientId, customChannelName)
+    if (cached) {
+      return cached
+    }
+
+    const rows = await this.query().where({ clientId, customChannelName })
```

Comment:
```suggestion
    const rows = await this.query().where({ clientId, customChannelName }).first()
```

Reply:
Not putting first here is consistent with the rest of the codebase. Anyways this query isn't supposed to return more than one row